### PR TITLE
Fix: Compare action should update its original comment instead of a new comment

### LINKED ISSUE #6047: Compare action should update its original comment instead of a new comment
As with https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5634 merged, we now have a script that compares any new assets added 🎉 
However, the script creates a new comment on each commit pushed, we would love if it just updates the top comment.

### PR DESCRIPTION
**Pull request by dinhtungdu**
Fixes #6047

This PR updates the compare assets actions to prevent it from generating a new comment each time developers push to a PR. Instead, the action will update the first report comment.

### Screenshots

Report comments are updated:

image

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

See the only one reported comment is created then edited for PRs that are pushed multiple times.

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Performance Impact

N/A

### Changelog

N/A

### CODE REVIEW

File: .github/compare-assets/index.js [hunk 1]
```diff
@@ -82,24 +82,55 @@ const runner = async () => {
...
-				'__This comment was automatically generated by the `./github/compare-assets` action.__',
 		} );
+
+		let reportCommentId;
+		if (
+			Array.isArray( currentComments.data ) &&
+			currentComments.data.length > 0
+		) {
+			const comment = currentComments.data.find( ( comment ) =>
+				comment.body.includes( 'Script Dependencies Report' )
```
Comment (senadir) [inline .github/compare-assets/index.js]: Minor concern is that if someone says this phrase that comment will be used, maybe we can add something hidden in the report and use that? That's not a blocker and I think this PR is great, Thank you!!

**Review by senadir — APPROVED** [review]

File: .github/compare-assets/index.js (hunk 1 above)
Comment (dinhtungdu) [reply .github/compare-assets/index.js]: > maybe we can add something hidden in the report and use that?

GitHub doesn't support storing metadata to its objects. Even if we can hack it to store metadata in the comment body, we're still parsing the comment body. It also makes the condition more complex, which isn't worth the effort IMO.

**Comment by dinhtungdu** [comment]
@senadir I made some updates to the action to ensure we only edit the dependencies report comment and cover the scenario when the dependencies were changed in former commits but then unchanged in the later ones. Can you please take another look?

You can check this comment for the history: https://github.com/woocommerce/woocommerce-blocks/pull/6736#issuecomment-1192461298

File: .github/compare-assets/index.js [hunk 2]
```diff
@@ -51,7 +51,45 @@ const runner = async () => {
...
+			} );
+
+			if (
+				Array.isArray( currentComments.data ) &&
+				currentComments.data.length > 0
+			) {
+				const comment = currentComments.data.find(
+					( comment ) =>
+						comment.body.includes( 'Script Dependencies Report' ) &&
+						comment.user.login === 'github-actions[bot]'
```
Comment (senadir) [inline .github/compare-assets/index.js:70]: Nice update!

**Review by senadir — APPROVED** [review]
This change is a well needed one, thank you for working on it Tung!

**Pull request merged**